### ISSUE TITLE
Load Image from Memory

### ISSUE THREAD

**Issue by ffreality** [libharu/libharu#273]
I am working on runtime PDF import and export plugin for Unreal Engine. I finished reader side with PDFium but I didn't like its writer features especially for images.

So, I want to use another library. Is there any way to insert images from memory but not file ?
I have BGRA Texture2D objects and I can get their buffer as void pointer. I can copy their datas to other image based objects with memcpy. Is there any way to do that in this library ?

**Comment by bramton** [comment]
Perhaps `HPDF_Image_LoadRawImageFromMem` can help you out? There is also a recent PR #271 which might be of interest to you.

**Comment by ffreality** [comment]
Yes, it worked. I didn't use any png or jpeg library and this functions can insert images with any extensions.

`	HPDF_Page Target_Page = HPDF_GetPageByIndex(In_PDF->Document, Page_Index);

	int32 Texture_Width = Target_Image->GetSizeX();
	int32 Texture_Height = Target_Image->GetSizeY();

	FTexture2DMipMap& Texture_Mip = Target_Image->GetPlatformData()->Mips[0];
	void* Texture_Data = Texture_Mip.BulkData.Lock(LOCK_READ_WRITE);
	
	TArray<FColor> Array_Colors;
	Array_Colors.SetNum(Texture_Width * Texture_Height);
	FMemory::Memcpy(Array_Colors.GetData(), static_cast<FColor*>(Texture_Data), static_cast<SIZE_T>(Array_Colors.Num()) * 4);

	Texture_Mip.BulkData.Unlock();
	Target_Image->UpdateResource();

	HPDF_BYTE* Buffer = (unsigned char*)malloc(static_cast<size_t>(Array_Colors.Num()) * 3);
	for (int32 Index_Colors = 0; Index_Colors < Array_Colors.Num(); Index_Colors++)
	{
		Buffer[Index_Colors * 3] = Array_Colors[Index_Colors].R;
		Buffer[(Index_Colors * 3) + 1] = Array_Colors[Index_Colors].G;
		Buffer[(Index_Colors * 3) + 2] = Array_Colors[Index_Colors].B;
	}
	
	HPDF_Image PDF_Image = HPDF_LoadRawImageFromMem(In_PDF->Document, Buffer, Texture_Width, Texture_Height, HPDF_ColorSpace::HPDF_CS_DEVICE_RGB, 8);
	HPDF_Page_DrawImage(Target_Page, PDF_Image, Position.X, Position.Y, Texture_Width, Texture_Height);`
